packages/app/src/client/services/PageContainer.js

```diff
@@ -282,11 +300,23 @@ export default class PageContainer extends Container {
   }
 
   async toggleLike() {
-    const bool = !this.state.isLiked;
-    await this.appContainer.apiv3Put('/page/likes', { pageId: this.state.pageId, bool });
-    this.setState({ isLiked: bool });
+    {
+      const bool = !this.state.isLiked;
+      await this.appContainer.apiv3Put('/page/likes', { pageId: this.state.pageId, bool });
+      this.setState({ isLiked: bool });
+
+      if (bool) {
+        this.setState({
+          likerIds: [...this.state.likerIds.split(',').filter(id => id !== ''), this.appContainer.currentUserId].join(','),
+        });
+      }
+      else {
+        this.setState({ likerIds: this.state.likerIds.split(',').filter(id => id !== '' && id !== this.appContainer.currentUserId).join(',') });
+      }
+    }
 
-    return this.retrieveLikeInfo();
+    await this.retrieveLikeInfo();
+    return this.retrieveLikers();
```

Comment:
add `likerIds` to the response of `/page/like-info` API instead of calling `/users.list`.